internal/storage/db.go

```diff
@@ -25,16 +34,339 @@ func InitDB(ctx context.Context, dsn string) (db *pgx.Conn, err error) {
+	if err != nil {
+		zap.S().Errorln("Insert order error: ", err)
+		return err
+	}
+
+	return nil
+}
+
+func (base *RepoMarket) IsExistForUser(ctx context.Context, userID *uuid.UUID, order string) (isExist bool, err error) {
+
+	var ordersn int
+	row := base.master.QueryRow(ctx, "SELECT count(*) FROM orders WHERE user_id = $1 AND onumber = $2", userID, order)
+	err = row.Scan(&ordersn)
+	if err != nil {
+		zap.S().Errorln("Error during order search for user: ", userID, order, err)
+		return true, err
+	}
+	if ordersn == 0 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func (base *RepoMarket) IsExistForOtherUsers(ctx context.Context, userID *uuid.UUID, order string) (isExist bool, err error) {
+	var ordersn int
+	row := base.master.QueryRow(ctx, "SELECT count(*) FROM orders WHERE user_id != $1 AND onumber = $2", userID, order)
+	err = row.Scan(&ordersn)
+	if err != nil {
+		zap.S().Errorln("Error during order search for user: ", userID, order, err)
+		return true, err
+	}
+	if ordersn == 0 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func (base *RepoMarket) IsPreOrder(ctx context.Context, userID *uuid.UUID, order string) (isPreOrder bool, err error) {
+	query := `
+	SELECT count(orders.onumber)
+	FROM orders INNER JOIN users ON orders.user_id = users.user_id 
+	WHERE users.user_id = $1 AND orders.onumber = $2 AND isPreorder = TRUE
+	`
+	row := base.master.QueryRow(ctx, query, userID, order)
+	var n int
+	err = row.Scan(&n)
+
+	return n == 1, err
+}
+
+func (base *RepoMarket) MovePreOrder(ctx context.Context, order *model.Order) (err error) {
+
+	_, err = base.master.Exec(ctx, "UPDATE orders SET status = $1, isPreorder = $2 WHERE onumber = $3", order.Status.String(), order.IsPreOrder, order.Onumber)
+	if err != nil {
+		zap.S().Errorln("UPDATE preoreder error: ", err)
+		return err
+	}
+
+	_, err = base.master.Exec(ctx, "UPDATE bonuses SET bonus_accrual = $1 WHERE onumber = $2", order.Bonus.Accrual, order.Onumber)
+	if err != nil {
+		zap.S().Errorln("UPDATE preoreder's bonus error: ", err)
+		return err
+	}
+
+	return
+
+}
+
+// Load all orders with not finished preparation status.
+func (base *RepoMarket) LoadPocessing(ctx context.Context) ([]model.Order, error) {
 
+	query := `
+	SELECT users.user_id, orders.onumber, orders.uploaded, orders.status
+	FROM orders INNER JOIN users ON orders.user_id = users.user_id 
+	WHERE (status = 'NEW' OR status = 'REGISTERED' OR status = 'PROCESSING') AND isPreorder = FALSE
+	`
+
+	rows, err := base.master.Query(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+	orders := []model.Order{}
+	for rows.Next() {
+		var order model.Order
+		var status string
+		err = rows.Scan(&order.UserID, &order.Onumber, &order.Uploaded, &status)
+		if err != nil {
+			return nil, err
+		}
+		order.Status.SetStatus(status)
+		orders = append(orders, order)
+	}
+
+	return orders, nil
+}
+
+func (base *RepoMarket) UpdateStatus(ctx context.Context, order *model.Order, accrual *decimal.Decimal) (err error) {
```

Comment:
Тут несколько проблем:
1 - мы не должны требовать лишние данные. Для смены статуса достаточно знать номер заказа и его статус. Все остальное избыточно и запрашивать не нужно. В разных ситуациях может просто не быть этих лишних данных и их придется где-то брать, хотя они и не нужны. Плюс это добавляет больший объем в стек, что тоже не нужно

2 - метод называется "обновить статус" - это хорошо и правильно. Но неправильно при этом еще "заодно" бонусный счет изменять. Клиентский код об этом не подозревает, это скрытое и неочевидное поведение. Лучше сделать отдельную функцию рядом
```suggestion
func (base *RepoMarket) UpdateStatus(ctx context.Context, orderNr string, orderStatus string) (err error) {
```

Reply:
Почистил методы на предмет передачи лишних параметров. Убрал скрытое поведение.